agent/bootstrap.go

```diff
@@ -121,14 +125,30 @@ func (boot bootstrap) setUserPasswords(env boshsettings.Env) error {
 		return nil
 	}
 
+	if !env.GetKeepRootPassword() {
+		err := boot.platform.SetUserPassword(boshsettings.RootUsername, password)
+		if err != nil {
+			return bosherr.WrapError(err, "Setting root password")
+		}
+	}
+
 	err := boot.platform.SetUserPassword(boshsettings.RootUsername, password)
 	if err != nil {
-		return bosherr.WrapError(err, "Setting root password")
+		return bosherr.WrapError(err, "Setting vcap password")
+	}
+
+	return nil
+}
+
+func (boot bootstrap) setRootPasswords(env boshsettings.Env) error {
```

Comment:
Yes, it's still working in progress. I will post a comment to ask you for review after I complete all code change and unit test. Thanks.